### ISSUE TITLE
Find some problems

### ISSUE THREAD

**Issue by qq1176914912** [simpleidserver/SimpleIdServer#794]
Hello, excuse me, I found a problem:
1、When I use https://localhost:5002/master/clients after the success of the login, just click exit, will stay on the page, And displays "Sorry, there's nothing at this address." This is the left function can be used normally, but in fact has been successfully exited：
![image](https://github.com/user-attachments/assets/e89bacef-db9b-4ac2-9b71-c02443acf265)
As before, clicking the exit button will jump to the 5001 system for exit operation.
I also try to your deployment project: https://website.simpleidserver.com/, click on the project, also won't jump, but shows "Sorry, there 's nothing at this address.".
2、As for the form above 5002, do you have an arrangement to add a search box? In order to deal with the large amount of data, convenient search and positioning?

**Comment by simpleidserver** [comment]
Indeed, the logout link is no longer working due to an issue in the 'RealmRouter.' :(

The changes have been published to the master branch. Could you try again?

I've created a new ticket (#795) to support 'advanced search' on the screen that contains a table.

**Comment by qq1176914912** [comment]
> Indeed, the logout link is no longer working due to an issue in the 'RealmRouter.' :(
> 
> The changes have been published to the master branch. Could you try again?
> 
> I've created a new ticket (#795) to support 'advanced search' on the screen that contains a table.

Thank you for your reply. I have tested it, and now it is possible to exit normally and jump to the login page, but the 'Manual action is required to revoke a session' configuration on the client side does not work. No matter whether it is checked or not, the page for manually ending a session will not appear.

**Comment by qq1176914912** [comment]
I found some new problems:
1、Users who only have view permissions can still do the following:
![image](https://github.com/user-attachments/assets/141472b5-fc5a-4e68-b755-e8cb32b643e4)
![image](https://github.com/user-attachments/assets/74399758-5697-4954-8a47-6e1af2423a0a)
Identity Scope Mappers of type User property and User attribute in Scope can be edited:
![image](https://github.com/user-attachments/assets/c3ec256b-fedd-43c2-b03a-8410da91193a)
"Registration" in acrs can also be modified normally, which should be disabled：
![image](https://github.com/user-attachments/assets/35fb9e63-9e96-4f4c-84d6-676e34193b27)
Start buttons in Scim and LDAP
![image](https://github.com/user-attachments/assets/4b905c06-a476-4f84-ae5f-42f03866382b)
Everything in Authentications can be updated:
![image](https://github.com/user-attachments/assets/ab21da21-ed82-40f6-9f4e-0e91dba782de)
![image](https://github.com/user-attachments/assets/429bab90-c95b-4b4d-9d7e-7a199024e3e0)
External identity providers in Authentications can also be updated:
![image](https://github.com/user-attachments/assets/a58f8ec0-94ee-4aa1-be05-7feb5c28058e)
The above content should not be accessible to users with view permissions only.
2、Go to the External identity providers menu in Authentications, and when editing the external login, the Properties and Mappers pages cannot be accessed:
![image](https://github.com/user-attachments/assets/292bb7d2-73e1-417e-8647-9717de5a1e26)
![image](https://github.com/user-attachments/assets/a93187b5-688a-454d-be9d-bfc34d7ac8db)
3、When I create a new realm in master, I switch to this realm, which takes me to the login page, and when I successfully log in to this realm, I click the Exit button, which takes me back to master instead of ending the session and taking me to the login page, and I have to click Exit again in master
4、The Extract in the new realm is empty, should SCIM and LDAP be added by default?
![image](https://github.com/user-attachments/assets/f6b4e1f7-09e6-4a38-a0ac-71cd72252225)
5、So let's say I have two realms, I log in to a master realm using administrator, and then I click Toggle realm to switch to a realm called testrealm, which will take me to the login page to log in again, I log in with the user name testuser3 (which doesn't exist in the master realm), and when I log in successfully, I switch back to the master realm, so I don't have to log in again. What's the logic here? The testuser3 user I logged in from testrealm doesn't exist in master. Should I log in again as well? Or will the tokens of the two users switch automatically when switching teams?

**Comment by qq1176914912** [comment]
6、The default administrator group created by Groups in the newly created realm is updated incorrectly
![image](https://github.com/user-attachments/assets/9ca28eb3-62de-4435-b488-9ee403f419b9)

**Comment by qq1176914912** [comment]
7、May be due to increased access problems now, visit https://localhost:5002/master/clients now, after login successfully, then the same browser to visit https://localhost:5001/master and log in again, There is no need to log in again without adding permissions.

**Comment by simpleidserver** [comment]
**The button to revoke the user's session is not displayed.**

This issue has been fixed in the master branch. The cookie name used by the administration website was incorrect.

**Users with READ permissions must not be able to update entities.**

This permission issue has been fixed in the master branch.

**Navigation is not working on the Identity Provider screen.**

This navigation issue has been fixed in the master branch.

**After logout, the user agent is redirected to the master realm.**

The redirection problem has been fixed in the master branch.

**Logging in with two different users in two different realms.**

If you log in to the master realm with the administrator user and to the testrealm realm with the testuser3 user, you will have two active sessions. It won't be necessary to authenticate again. You’ll automatically be authenticated with the correct user based on the realm.

**Group update date and time are invalid.**

This issue has been fixed in the master branch.

**After logging in to https://localhost:5002/master/clients and browsing the website https://localhost:5001/master, the login window is displayed.**

This issue has also been fixed in the master branch.

**Comment by qq1176914912** [comment]
> **The button to revoke the user's session is not displayed.**
> 
> This issue has been fixed in the master branch. The cookie name used by the administration website was incorrect.
> 
> **Users with READ permissions must not be able to update entities.**
> 
> This permission issue has been fixed in the master branch.
> 
> **Navigation is not working on the Identity Provider screen.**
> 
> This navigation issue has been fixed in the master branch.
> 
> **After logout, the user agent is redirected to the master realm.**
> 
> The redirection problem has been fixed in the master branch.
> 
> **Logging in with two different users in two different realms.**
> 
> If you log in to the master realm with the administrator user and to the testrealm realm with the testuser3 user, you will have two active sessions. It won't be necessary to authenticate again. You’ll automatically be authenticated with the correct user based on the realm.
> 
> **Group update date and time are invalid.**
> 
> This issue has been fixed in the master branch.
> 
> **After logging in to https://localhost:5002/master/clients and browsing the website https://localhost:5001/master, the login window is displayed.**
> 
> This issue has also been fixed in the master branch.

Thanks for your reply, these problems have been solved after testing

**Comment by qq1176914912** [comment]
Sorry to bother you, but I found a new problem:
1. Select the Identity Resource type in the Scopes. After entering the Mappers, select User property. There is no response when you click the first time, and the page keeps loading after the second click. Can I edit the maaper whose type is User property?
2. After clicking the Add client role button in Groups, as long as the Role page is expanded, that is, the permission page, and then click the Scope menu, the content of the permission will be replaced with Scope
3. External identity providers in Authentications, select one to edit,
Go to Mappers and select USERATTRIBUTE to edit it. There is also a save button if users only have viewing rights. When the USERPROPERTY type is selected for editing, there is no save button, even for users with edit permissions, and once entered, there is no way to go back to the previous page
4. Extract Mapping rules option, only users with viewing permissions still show the save button, that is, you can edit it
[view.zip](https://github.com/user-attachments/files/17484673/view.zip)
The first three questions have been recorded on video for your convenience

**Comment by qq1176914912** [comment]
5、Registration Workflows when a new registration Is added, if 'Is default? 'After the creation Is successful, the table will not cancel the previous Is default, resulting in two checked states:
![image](https://github.com/user-attachments/assets/de857b9f-9563-48af-b5d2-35ab58f741f6)
![image](https://github.com/user-attachments/assets/7378cb25-5f36-4748-8bee-97dde645dab9)

**Comment by simpleidserver** [comment]
All the issues are fixed in the `master` branch :)

**Comment by qq1176914912** [comment]
> All the issues are fixed in the `master` branch :) can-you try again?

Thank you for your reply. One more issue, only users with view permissions still have save permissions in Extract->LDAP/SCIM->Mapping rules.

https://github.com/user-attachments/assets/8a50b94e-94a1-41e0-b9f1-f6c54c18276f

**Comment by simpleidserver** [comment]
The issue is fixed in the `master` branch.

**Comment by qq1176914912** [comment]
> The issue is fixed in the `master` branch.

Thank you for your reply, the problems found so far have been resolved

**Comment by simpleidserver** [comment]
The issue is fixed in the `master` branch.

**Comment by qq1176914912** [comment]
> The issue is fixed in the `master` branch.

The Scopes menu also has this problem
![image](https://github.com/user-attachments/assets/c4c4d32c-a844-42a2-92e6-8b3e361163c3)

**Comment by simpleidserver** [comment]
The issue is fixed in the `master` branch.

**Comment by qq1176914912** [comment]
I found that some pages can be updated normally when entering the edit for the first time by clicking Update, but after F5 refreshes the browser on this page, it cannot be saved normally, and sometimes it will lead to the entire menu card master
[view.zip](https://github.com/user-attachments/files/17585959/view.zip)

**Comment by simpleidserver** [comment]
Hello,

The issues have been fixed in the Release503 branch. All the NuGet packages referenced by the SimpleIdServer project have also been updated. You might notice some regressions in the look and feel of the UI due to significant changes made in the latest version of the Blazor library.

Could you please fetch the latest changes and try again?

KR,
SID

**Comment by qq1176914912** [comment]
> Hello,
> 
> The issues have been fixed in the Release503 branch. All the NuGet packages referenced by the SimpleIdServer project have also been updated. You might notice some regressions in the look and feel of the UI due to significant changes made in the latest version of the Blazor library.
> 
> Could you please fetch the latest changes and try again?
> 
> KR, SID

Thank you for your reply. Is Release503 the same as the master branch except that it updates the nuget package and the previous problem?

**Comment by qq1176914912** [comment]
There is also a page not refresh issue:
The page of Auditing will load data only when it is opened for the first time. After an addition operation is performed, the log content will not appear if you click the Auditing again. It will be refreshed only after the browser is refreshed

https://github.com/user-attachments/assets/5ba83ee4-4aef-431b-b78a-e606317ede29

**Comment by simpleidserver** [comment]
Hello,

It is normal that the log "The client is registered" does not appear on the screen immediately because the integration event "ClientRegisteredSuccessEvent" is processed by the in-memory bus after a certain delay.

Regarding your CSS issue, did you create your own CSS file, or are you using the default one provided by SimpleIdServer.

**Comment by qq1176914912** [comment]
> Hello,
> 
> It is normal that the log "The client is registered" does not appear on the screen immediately because the integration event "ClientRegisteredSuccessEvent" is processed by the in-memory bus after a certain delay.
> 
> Regarding your CSS issue, did you create your own CSS file, or are you using the default one provided by SimpleIdServer.

I downloaded your project directly and started it after all the regeneration. Do you mean that it is not like this in your place?I downloaded your project directly and started it after all the regeneration. Do you mean that it is not like this in your place? I'll try again later

**Comment by qq1176914912** [comment]
> > Hello,
> > It is normal that the log "The client is registered" does not appear on the screen immediately because the integration event "ClientRegisteredSuccessEvent" is processed by the in-memory bus after a certain delay.
> > Regarding your CSS issue, did you create your own CSS file, or are you using the default one provided by SimpleIdServer.
> 
> I downloaded your project directly and started it after all the regeneration. Do you mean that it is not like this in your place?I downloaded your project directly and started it after all the regeneration. Do you mean that it is not like this in your place? I'll try again later

I tried again, regenerated the project and started, and then the browser visited ctrl+f5 after refresh, it is like that, if you do not have this is not your browser has a cache?

**Comment by simpleidserver** [comment]
Can-you replace the content of the `_Host.cshtml` file with the following content and try again ?

```
﻿@page "/"
@using Microsoft.AspNetCore.Components.Web
@namespace SimpleIdServer.IdServer.Website.Startup.Pages
@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <base href="~/" />
    <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/bootstrap/bootstrap.min.css" />
    <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/website.css" />
    <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/stateDig.css" />
    <link rel="icon" type="image/png" href="_content/SimpleIdServer.IdServer.Website/favicon.png" />
    <script src="_content/SimpleIdServer.IdServer.Website/website.js"></script>
    <script src="_content/Radzen.Blazor/Radzen.Blazor.js"></script>
    <component type="typeof(HeadOutlet)" render-mode="ServerPrerendered" />
</head>
<body class="orange">
    <component type="typeof(App)" render-mode="ServerPrerendered" />

    <div id="blazor-error-ui">
        <environment include="Staging,Production">
            An error has occurred. This application may no longer respond until reloaded.
        </environment>
        <environment include="Development">
            An unhandled exception has occurred. See browser dev tools for details.
        </environment>
        <a href="" class="reload">Reload</a>
        <a class="dismiss">🗙</a>
    </div>

    <script src="_framework/blazor.server.js"></script>
</body>
</html>
```

**Comment by qq1176914912** [comment]
> Can-you replace the content of the `_Host.cshtml` file with the following content and try again ?
> 
> Source code : https://raw.githubusercontent.com/simpleidserver/SimpleIdServer/refs/heads/master/src/IdServer/SimpleIdServer.IdServer.Website.Startup/Pages/_Host.cshtml
> 
> ```
> ﻿@page "/"
> @using Microsoft.AspNetCore.Components.Web
> @namespace SimpleIdServer.IdServer.Website.Startup.Pages
> @addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
> 
> <!DOCTYPE html>
> <html lang="en">
> <head>
>     <meta charset="utf-8" />
>     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
>     <base href="~/" />
>     <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/bootstrap/bootstrap.min.css" />
>     <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/website.css" />
>     <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/stateDig.css" />
>     <link rel="icon" type="image/png" href="_content/SimpleIdServer.IdServer.Website/favicon.png" />
>     <script src="_content/SimpleIdServer.IdServer.Website/website.js"></script>
>     <script src="_content/Radzen.Blazor/Radzen.Blazor.js"></script>
>     <component type="typeof(HeadOutlet)" render-mode="ServerPrerendered" />
> </head>
> <body class="orange">
>     <component type="typeof(App)" render-mode="ServerPrerendered" />
> 
>     <div id="blazor-error-ui">
>         <environment include="Staging,Production">
>             An error has occurred. This application may no longer respond until reloaded.
>         </environment>
>         <environment include="Development">
>             An unhandled exception has occurred. See browser dev tools for details.
>         </environment>
>         <a href="" class="reload">Reload</a>
>         <a class="dismiss">🗙</a>
>     </div>
> 
>     <script src="_framework/blazor.server.js"></script>
> </body>
> </html>
> ```

I downloaded the project of your Release503 branch, and replaced the content of the "_host-cshtml" file with the content you said according to what you said, but it still cannot be displayed normally, and the content cannot be loaded into a menu.
![image](https://github.com/user-attachments/assets/33eea378-2c7a-499f-8649-384baf4e70d0)

**Comment by simpleidserver** [comment]
Oups, i made a mistake. The correct `_Host.cshtml` file has the following content  : 

Source code : https://github.com/simpleidserver/SimpleIdServer/blob/Release503/src/IdServer/SimpleIdServer.IdServer.Website.Startup/Pages/_Host.cshtml

```
@page "/"
@using Microsoft.AspNetCore.Components.Web
@namespace SimpleIdServer.IdServer.Website.Startup.Pages
@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <base href="~/" />
    <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/bootstrap/bootstrap.min.css" />
    <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/themes.css" />
    <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/website.css" />
    <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/stateDig.css" />
    <link rel="icon" type="image/png" href="_content/SimpleIdServer.IdServer.Website/favicon.png" />
    <script src="_content/SimpleIdServer.IdServer.Website/website.js"></script>
    <component type="typeof(HeadOutlet)" render-mode="ServerPrerendered" />
</head>
<body class="orange">
    <component type="typeof(App)" render-mode="ServerPrerendered" />

    <div id="blazor-error-ui">
        <environment include="Staging,Production">
            An error has occurred. This application may no longer respond until reloaded.
        </environment>
        <environment include="Development">
            An unhandled exception has occurred. See browser dev tools for details.
        </environment>
        <a href="" class="reload">Reload</a>
        <a class="dismiss">🗙</a>
    </div>

    <script src="_framework/blazor.server.js"></script>
    <script src="_content/Radzen.Blazor/Radzen.Blazor.js"></script>
</body>
</html>
```

**Comment by qq1176914912** [comment]
> Oups, i made a mistake. The correct `_Host.cshtml` file has the following content :
> 
> Source code : https://github.com/simpleidserver/SimpleIdServer/blob/Release503/src/IdServer/SimpleIdServer.IdServer.Website.Startup/Pages/_Host.cshtml
> 
> ```
> @page "/"
> @using Microsoft.AspNetCore.Components.Web
> @namespace SimpleIdServer.IdServer.Website.Startup.Pages
> @addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
> 
> <!DOCTYPE html>
> <html lang="en">
> <head>
>     <meta charset="utf-8" />
>     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
>     <base href="~/" />
>     <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/bootstrap/bootstrap.min.css" />
>     <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/themes.css" />
>     <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/website.css" />
>     <link rel="stylesheet" href="_content/SimpleIdServer.IdServer.Website/css/stateDig.css" />
>     <link rel="icon" type="image/png" href="_content/SimpleIdServer.IdServer.Website/favicon.png" />
>     <script src="_content/SimpleIdServer.IdServer.Website/website.js"></script>
>     <component type="typeof(HeadOutlet)" render-mode="ServerPrerendered" />
> </head>
> <body class="orange">
>     <component type="typeof(App)" render-mode="ServerPrerendered" />
> 
>     <div id="blazor-error-ui">
>         <environment include="Staging,Production">
>             An error has occurred. This application may no longer respond until reloaded.
>         </environment>
>         <environment include="Development">
>             An unhandled exception has occurred. See browser dev tools for details.
>         </environment>
>         <a href="" class="reload">Reload</a>
>         <a class="dismiss">🗙</a>
>     </div>
> 
>     <script src="_framework/blazor.server.js"></script>
>     <script src="_content/Radzen.Blazor/Radzen.Blazor.js"></script>
> </body>
> </html>
> ```
Where do you show no problem?
It seems that website.css is not found. The Release503 branch project I downloaded does not contain website.css
![image](https://github.com/user-attachments/assets/8c6c9cce-6abb-4802-a99c-5be0f40749e2)
![image](https://github.com/user-attachments/assets/ee5fd472-5a3d-4b29-8fc8-c17440145202)

**Comment by simpleidserver** [comment]
I pushed the missing css file in the branch Release503.
Can-you try again?

**Comment by qq1176914912** [comment]
> I pushed the missing css file in the branch Release503. Can-you try again?

There is no problem, it can display normally now
![image](https://github.com/user-attachments/assets/bcf28626-045b-4524-9c78-770bf689f792)

**Comment by qq1176914912** [comment]
ACRS Questions:
When I added an authentication method called testacrpwd to master and bound the registration method to pwd, I bound this method to the 5002 client. At this point, the registration of the login page would work. Then I created a new realm and logged in. Click on it and say no, then log in. When you go to the new realm, you can see that acrs inherits the previous one, and then create a new registration method (name is different from the master). Then we bind the registration method to it, and the new realm registration will work, but the master registration will not work.
Here's a video of the operation:
[quenew.zip](https://github.com/user-attachments/files/17671909/quenew.zip)

**Comment by qq1176914912** [comment]
2、If you enter the wrong password or account number, the sign up button will disappear

https://github.com/user-attachments/assets/3239237e-c80a-45e0-8fb5-a1f44fc2f26a

**Comment by qq1176914912** [comment]
3、I had a situation where sometimes when toggled back and forth between realm login, the browser was prompted with "HTTP ERROR 431", I looked it up and explained: "When you send a request to the server, the HTTP request contains some header fields (such as cookies, User-Agent, Accept, etc.). These fields have a certain size limit. If the header field of the request is too large, the server returns a 431 error indicating that the header of the request cannot be processed because it exceeds the size allowed by the server." I don't know if you've ever encountered it.

**Comment by simpleidserver** [comment]
Hello,

The following issues have been fixed in the Release503 branch:

* Fixed issues with ACR.
* Ensured the Register button is always displayed.
* All client roles are now displayed, and pagination is active.

Regarding the HTTP 431 error, this exception is thrown because there are too many active sessions, causing the cookie header size to exceed the limit. 
I have created ticket #807 to limit the number of active sessions and display them across the different UIs.

**Comment by simpleidserver** [comment]
The ticket #807 has been developed, and the code has been pushed to the Release503 branch. 
Therefore, the HTTP error 431 should no longer occur. 
By default, the number of active sessions is limited to 4, defined by the `MaxNbActiveSessions` property in IdServerHostOptions, which can be adjusted as needed.

**Comment by simpleidserver** [comment]
When there is a limit of 4 sessions, the fifth session will not replace an active session. 
Instead, an error message will be displayed in the UI, and the end-user will be responsible for revoking a session via the URL: https://localhost:5001.

**Comment by qq1176914912** [comment]
> When there is a limit of 4 sessions, the fifth session will not replace an active session. Instead, an error message will be displayed in the UI, and the end-user will be responsible for revoking a session via the URL: https://localhost:5001.

Thanks for your reply. Got it
